### PR TITLE
overlord, tests: add managers and a spread test for UC20 to UC22 remodel

### PR DESCRIPTION
**Pull request by bboozzoo**
Based on top of #11102

### CODE REVIEW

**Review by mardy — APPROVED** [review]
LGTM, FWIW :-)

File: tests/lib/assertions/valid-for-testing-pc-22-from-20.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            "type": "kernel"
+        },
+        {
+            "default-channel": "latest/edge",
+            "id": "amcUKQILKXHHTlmSa7NMdnXSx02dNeeT",
+            "name": "core22",
+            "type": "base"
+        },
+        {
+            "default-channel": "latest/stable",
```
Comment (anonymouse64) [inline tests/lib/assertions/valid-for-testing-pc-22-from-20.json:32]: not that it really matters in the slightest, but shouldn't this be edge to match the other snaps ?

```suggestion
            "default-channel": "latest/edge",
```

File: tests/nested/manual/core20-to-core22/task.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+# the test may be unstable as UC22 is effectively a work-in-progress thing, and
+# the model in question uses latest/edge of core22 and 22/edge of pc and
+# pc-kernel snaps
+
+systems: [ubuntu-20.04-64]
+
+environment:
+  NESTED_CUSTOM_MODEL: $TESTSLIB/assertions/valid-for-testing-pc-20.model
+  NESTED_IMAGE_ID: uc22-remodel-testing
+  # investigate why system cannot be booted after remodel completes
```
Comment (anonymouse64) [inline tests/nested/manual/core20-to-core22/task.yaml:12]: what does this comment mean? is this an old comment? or are we not able to boot the system if we use FDE and we do the remodel?

**Review by anonymouse64 — APPROVED** [review]
lgtm, a couple questions

File: tests/nested/manual/core20-to-core22/task.yaml (hunk 2 above)
Comment (bboozzoo) [reply tests/nested/manual/core20-to-core22/task.yaml:12]: I'll make this a TODO, but yes there appears to be a problem when booting with FDE which needs investigation:
0LjNNyT

Comment (anonymouse64) [reply tests/nested/manual/core20-to-core22/task.yaml]: ooooh nice that's a new problem from snap-bootstrap how lovely

**Pull request merged**